src/main/java/com/github/junit5docker/DefaultDockerClient.java

```diff
@@ -25,12 +30,67 @@
     }
 
     @Override
-    public String startContainer(String wantedImage, Map<String, String> environment, PortBinding... portBinding) {
+    public ContainerInfo startContainer(String wantedImage,
+                                        Map<String, String> environment,
+                                        String[] networkNames,
+                                        PortBinding... portBinding) {
         Ports bindings = createPortBindings(portBinding);
         List<String> environmentStrings = createEnvironmentList(environment);
         String containerId = createContainer(wantedImage, bindings, environmentStrings);
         dockerClient.startContainerCmd(containerId).exec();
-        return containerId;
+        Collection<String> networkIds = joinNetworks(networkNames, containerId);
+        return new ContainerInfo(containerId, networkIds);
+    }
+
+    private Collection<String> joinNetworks(String[] networkNames, String containerId) {
+        if (networkNames == null || networkNames.length == 0) {
+            return Collections.emptySet();
+        }
+        Collection<String> networkIds = new HashSet<>();
+
+        for (String network : networkNames) {
+            String networkId = getNetworkId(network);
+            dockerClient.connectToNetworkCmd()
+                        .withNetworkId(networkId)
+                        .withContainerId(containerId)
+                        .exec();
+            networkIds.add(networkId);
+        }
+        return networkIds;
+    }
+
+    private String getNetworkId(String networkName) {
+        return getExistingNetworkId(networkName)
+                   .orElseGet(() -> dockerClient.createNetworkCmd().withName(networkName).exec().getId());
+    }
+
+    private Optional<String> getExistingNetworkId(String networkName) {
+        return dockerClient.listNetworksCmd()
+                           .exec()
+                           .stream()
+                           .filter(name -> name.getName().equals(networkName))
+                           .reduce((name1, name2) -> {
+                               String msg = String.format("Multiple networks found with the same name: %s and %s",
+                                                          name1,
+                                                          name2);
+                               throw new IllegalStateException(msg);
```

Comment:
Is it really something that can happen or is it a docker bug ?
If it's a docker bug, I think we could just get the first network having the right name. What do you think ?

Reply:
Hmm for some reason i thought i saw this happening during development when the networks weren't being cleaned up correctly so I added this guard. 
But testing it again now it seems you're right, that you can't have multiple networks with the same name. So i'll change it back to find first :)

Reply:
@hanleyt @FaustXVI I think there is a case you can if they don't have the same scope — i.e. one is local (bridge) and the other would be swarm (overaly — created elsewhere on the cluster). That said, I think it's an edge case and shouldn't really happen that much 😓

Reply:
@vdemeester well, I think it's the kind of edge case we actually don't want to handle. I think that when you start to have this kind of configuration, you're not really on a development machine (dev or ci). So maybe we should actually crash at the network creation time if the network already exists. What do you think ? or we can avoid the crash by appending a random value to it ?

Reply:
Both would work (i.e. let the thing crash at network creation, *or* adding a random value to make sure it's unique)